### ISSUE TITLE
P1: Qwen3 observer recomputes identical attention mask for every layer (should be hoisted out of loop)

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/reap-mlx#35]
## Summary

In `_observe_qwen3_model()`, the attention mask is computed **inside the layer loop**, causing `num_layers × num_sequences` mask creations instead of `num_sequences`.

## Location

`src/reap/observer.py`, lines ~87-96:

```python
for sequence in calibration_sequences:
    tokens = _batch_tokens(mx, sequence)
    h = embed_tokens(tokens)

    for layer_idx, layer in enumerate(layers):            # ← loop
        mask = _attention_mask(                            # ← INSIDE loop
            h, sequence_length=tokens.shape[-1],
            mask_fn=mask_fn,
        )
        h = _run_attention(layer, h, mask)
```

When `mask_fn` is `None` (the default), `_attention_mask()` calls `make_attention_mask(hidden_states, cache=None)`. Since `h` maintains the same shape `[1, seq_len, hidden]` across all decoder layers, the causal attention mask is **identical for every layer**.

The LFM2 observer already handles this correctly — masks are computed once per sequence outside the layer loop (lines ~142-146).

## Impact

For a 32-layer model with 8 calibration sequences at 1024 tokens: **~256 redundant mask allocations** instead of 8.

## Proposed fix

Hoist the default mask computation outside the layer loop, but preserve the `mask_fn` path inside the loop (since custom mask functions may depend on hidden state values):

```python
for sequence in calibration_sequences:
    tokens = _batch_tokens(mx, sequence)
    h = embed_tokens(tokens)
    default_mask = (
        _attention_mask(h, sequence_length=tokens.shape[-1], mask_fn=mask_fn)
        if mask_fn is None
        else None
    )

    for layer_idx, layer in enumerate(layers):
        mask = mask_fn(h, cache=None) if mask_fn is not None else default_mask
        h = _run_attention(layer, h, mask)
        ...
```

**Effort**: ~5 lines in `observer.py`.
